### ISSUE TITLE
Missing type for `processor` in `ReplaceInFileConfig`

### ISSUE THREAD

**Issue by julrich** [adamreisnz/replace-in-file#170]
I think this one is just plain missing, right? Same for the array version of it?

https://github.com/adamreisnz/replace-in-file#custom-processor
https://github.com/adamreisnz/replace-in-file/blob/main/types/index.d.ts#L20-L31

... and after a small patch attempt locally... probably also need to make `from` and `to` optional, then?
Maybe this could be done more elegantly with some type of discriminator?

<img width="220" alt="image" src="https://github.com/adamreisnz/replace-in-file/assets/136760/4138eea5-31b6-44c1-8ea0-17035cbdaff4">

This looks interesting: https://stackoverflow.com/a/49265354

**Comment by adamreisnz** [comment]
Hi, if you are referring to typescript types, I haven't been managing those, but you're welcome to submit a PR to fix if needed. Thanks!

**Comment by julrich** [comment]
Cool, I'll have a look! Who maintained these before, do you know? Maybe makes sense to coordinate. If they're just not really maintained at all currently... it should be a quick fix!

**Comment by adamreisnz** [comment]
I am not sure, you could look at the past PR's related to typescript. Not really maintained though, so you're welcome to take charge!

**Comment by julrich** [comment]
Cool, I'll have a go, then!

**Comment by lodossDev** [comment]
hi, what happened to this? i am not sure how to fix this using typescript?

**Comment by julrich** [comment]
Hey @lodossDev, we're still living with a local patch for this, as I haven't found the time to investigate this more in-depth. Don't want to contribute something that's working for us, but nobody else :D

If you're feeling experimental, you can try the patch we're currently using:
```diff
diff --git a/node_modules/replace-in-file/types/index.d.ts b/node_modules/replace-in-file/types/index.d.ts
index 97409d5..bd88df2 100644
--- a/node_modules/replace-in-file/types/index.d.ts
+++ b/node_modules/replace-in-file/types/index.d.ts
@@ -1,7 +1,5 @@
 
 declare module 'replace-in-file' {
-  export function replaceInFile(config: ReplaceInFileConfig): Promise<ReplaceResult[]>;
-  export function replaceInFile(config: ReplaceInFileConfig, cb: (error: Error, results: ReplaceResult[]) => void): void;
   export default replaceInFile;
 
   namespace replaceInFile {
@@ -11,9 +9,6 @@ declare module 'replace-in-file' {
     export function replaceInFile(config: ReplaceInFileConfig, cb: (error: Error, results: ReplaceResult[]) => void): void;
   }
 
-  export function sync(config: ReplaceInFileConfig): ReplaceResult[];
-  export function replaceInFileSync(config: ReplaceInFileConfig): ReplaceResult[];
-
   export type From = string | RegExp | FromCallback;
   export type To = string | ToCallback;
```

We're applying it using https://www.npmjs.com/package/patch-package

**Comment by zach-babb-ef** [comment]
I have a pr for this, but not permissions to the repo if someone wants to add me. Fwiw OP is correct. I also cleaned up the syntax of the type to make it consistent.

**Comment by adamreisnz** [comment]
You can simply submit your PR and I will review/approve it, thank you!

**Comment by julrich** [comment]
Thank you a lot, @zach-babb-ef :raised_hands:

**Comment by adamreisnz** [comment]
Thank you, merged and release in 7.2.0
